src/store/soundsStore.js

```diff
@@ -48,16 +50,31 @@ const mutations = {
 		state.playSoundsUser = enabled
 		state.playSoundsGuest = enabled
 	},
+
+	setUserId(state, userId) {
+		state.userId = userId
+	},
 }
 
 const actions = {
+
+	/**
+	 * @param {object} context default store context;
+	 * @param {object} user A NextcloudUser object as returned by @nextcloud/auth
+	 * @param {string} user.uid The user id of the user
+	 */
+	setCurrentUser(context, user) {
+		context.commit('setUserId', user.uid)
```

Comment:
this will boom if user is null when dealing with guests ?

Reply:
https://github.com/nextcloud/spreed/blob/a661a156a0a6f2696db15cbad2b678807b5b779a/src/App.vue#L319-L324

Actor store also already depends on valid values: https://github.com/nextcloud/spreed/blob/6184127a50ad62e429600e2d6cc0310d2873af6d/src/store/actorStore.js#L124-L129 doc block is also rather clear it does not allow null

Reply:
yeah, while writing tests I remember that to set the user to null you need to call `setCurrentParticipant`.
it's a bit twisted, but explains it. let's move forward